### PR TITLE
mc_pos_control: Fix bug causing flyaway when GPS gained after takeoff

### PR DESCRIPTION
**Pull request by ChristophTobler**
The change in origin when GPS was gained after takeoff was being used to shift the set point despite the previous origin being invalid.
@priseborough FYI

### CODE REVIEW

File: src/modules/mc_pos_control/mc_pos_control_main.cpp [hunk 1]
```diff
@@ -856,7 +858,7 @@ MulticopterPositionControl::task_main_trampoline(int argc, char *argv[])
 void
 MulticopterPositionControl::update_ref()
 {
-	if (_local_pos.ref_timestamp != _ref_timestamp) {
+	if (_local_pos.ref_timestamp != _ref_timestamp && _first_origin_set) {
```
Comment (ChristophTobler) [inline src/modules/mc_pos_control/mc_pos_control_main.cpp]: @priseborough This PR just works for the situation where we use flow at the beginning and then gain GPS. The reason is that this line will never be fully true. If `_first_origin_set` is true, `_ref_timestamp` has been set to `_local_pos.ref_timestamp`. Hence we will never enter this code block.
For the normal simulation (`make posix_sitl_default gazebo_iris`) the quad climbs into the sky when you run `commander takeoff`.

**Comment by ChristophTobler** [comment]
The long term fix would be to move to local coordinates only, right? I think, we need to push for that.

File: src/modules/mc_pos_control/mc_pos_control_main.cpp (hunk 1 above)
Comment (priseborough) [reply src/modules/mc_pos_control/mc_pos_control_main.cpp]: OK, I will review and fix tomorrow.

**Comment by priseborough** [comment]
Agree - that will solve this type of problem, but would require the waypoints to be converted to local coordinates. It would also avoid the need for the expensive local to global transformation needing to be done at every time step. It would only need to be done once for each waypoint.

**Comment by ChristophTobler** [comment]
>Agree - that will solve this type of problem, but would require the waypoints to be converted to local coordinates.

@Stifael and I once started to work on that https://github.com/PX4/Firmware/pull/7432

**Comment by priseborough** [comment]
The last commit is the simplest way I can think of to prevent the reference point changing during flight, which is not safe with the way the controller works in global coordinates. 

This change is compatible with the ecl/EKF because it will only set the origin once power cycle. I've checked LPE and also appears to only set the origin once (@dagar can you confirm this?).

If it is absolutely necessary to retain the ability to change reference point in flight, then we can modify the logic to only allow it when in a waypoint driven flight mode.

**Comment by ChristophTobler** [comment]
@priseborough This seems to work for the EKF2! :+1:

@jgoppert Controller-wise it also seems to work for the LPE -> no jumps in setpoints. However, I see a jump in the estimated position when switching to use GPS as well (see below). Any ideas why that happens?
lpe_flow_to_gps

**Pull request merged**